### PR TITLE
[BPF] no need for src fixup for packets from wl to tunnel

### PR DESCRIPTION
**Pull request by tomastigera**
Previously, we had to preserve the pod's source on the already encapped vxlan packet when returning to the tunnel because when FIB failed, the packet was dropped to host namespace and strict RPF on the pod's iface would drop it.

We no longer need that because (1) RPF is now relaxed since BPF enforces strict on its own as there are other cases when a pod replies with service IP and (2) we forward 100% of the packets straight to the external interface (from where the traffic originally came) to avoid issues with source routing in EKS (which no longer is a problem with the non-pod's source)

## Related issues/PRs
refs https://github.com/projectcalico/calico/issues/7040

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: no src fixup on host iface for traffic returning from pod to the nodeport tunnel
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/fv/ipip_test.go
Comment (fasaxc) [inline felix/fv/ipip_test.go:422]: `Succeed()` TIL you can do `Should(Succeed())` instead of `ShouldNot(HaveOccurred())`

File: felix/fv/ipip_test.go
Comment (fasaxc) [inline felix/fv/ipip_test.go:446]: The key thing that we were testing was that, without an entry in the IP set, IPIP should be blocked.  So, suggest you add a couple of lines to this test that check that removing the IP from the config breaks connectivity.

**Review by fasaxc — CHANGES_REQUESTED** [review]
Yep, adding the ext client was what I expected, just think we need the negative aspect of the test too to really verify the IP set is used to block unexpected IPIP.

File: felix/fv/ipip_test.go
Comment (tomastigera) [reply felix/fv/ipip_test.go:446]: sure I can add that :+1:

File: felix/fv/ipip_test.go
Comment (fasaxc) [inline felix/fv/ipip_test.go:429]:
```suggestion
			By("changing configuration to include the external client")
```

**Review by fasaxc — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
